File: python/ui/ConverterFrame.py
import Tkinter as tk
from Tkinter import Frame
from ttk import Notebook
import tkFileDialog
import Tkconstants
import tkMessageBox

from NumbersPane import NumbersPane
from RawTextPane import RawTextPane

class ConverterFrame(Frame):
    def __init__(self,converter,master=None):
        Frame.__init__(self,master)

        master.title("Ai Variable tool")

        self.converter = converter

        self.grid(row=0, sticky=(tk.N+tk.E+tk.W+tk.S))
        self.grid_columnconfigure(0, weight=1)
        self.grid_rowconfigure(0, weight=1)

        self.tabbedPane = Notebook(self,name="tabbed-pane")
        self.tabbedPane.grid(row=0, column=0, sticky=(tk.N+tk.E+tk.W+tk.S))

        self.btnSave = tk.Button(self,text="Guardar",command=self.btnSaveCallback)
        self.btnSave.grid(row=1,column=0,sticky=(tk.N+tk.E+tk.W+tk.S))

        self.rawTextPanel = RawTextPane(self.tabbedPane)
        self.rawTextPanel.grid(row=0, column=0, sticky=(tk.N+tk.E+tk.W+tk.S))

        self.numbersPanel = NumbersPane(self.tabbedPane)
        self.numbersPanel.grid(row=0, column=0, sticky=(tk.N+tk.E+tk.W+tk.S))

        self.tabbedPane.add(self.numbersPanel,text="Series de numeros")
        self.tabbedPane.add(self.rawTextPanel, text="Texto en crudo")

        self.panels = [self.numbersPanel,self.rawTextPanel]

    def btnSaveCallback(self):        
        filename = tkFileDialog.asksaveasfilename(initialdir=".", title="Select file", filetypes=(("Xml files", "*.xml"),("All files","*.*")))

        current_idx = self.tabbedPane.index("current")
        work_text = self.panels[current_idx].getText()

        self.converter.convert(work_text)
        if not self.converter.save(filename):
            tkMessageBox.showerror("Error","El archivo no ha sido almacenado.")
        else:
            tkMessageBox.showinfo("Exito","El archivo ha sido almacenado adecuadamente.")

File: java/ui/ConverterFrame.java
package ui;

import java.io.File;
import java.awt.event.ActionListener;
import java.awt.event.ActionEvent;
import java.awt.GridBagLayout;
import java.awt.GridBagConstraints;
import javax.swing.JFrame;
import javax.swing.JButton;
import javax.swing.JTabbedPane;
import javax.swing.JFileChooser;
import javax.swing.JOptionPane;

import converter.Converter;

public class ConverterFrame extends JFrame implements ActionListener {
	private Converter converter;

	private JButton btnSave;
	private JTabbedPane tabbedPane;
	private GridBagLayout layout;

	private RawTextPane rawTextPanel;
	private NumbersPane numbersPanel;

	private ConverterFrame(){
		super("Ai Variable tool");
		setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);

		layout = new GridBagLayout();
		setLayout(layout);

		GridBagConstraints c = new GridBagConstraints();

		c.fill=GridBagConstraints.BOTH;
		c.gridx=0;
		c.gridy=0;
		c.gridwidth=1;
		tabbedPane = new JTabbedPane();
		rawTextPanel = new RawTextPane();
		numbersPanel = new NumbersPane();
		tabbedPane.add(numbersPanel,"Series de numeros");
		tabbedPane.add(rawTextPanel,"Texto en crudo");
		add(tabbedPane,c);
		

		c.fill=GridBagConstraints.HORIZONTAL;
		c.gridx=0;
		c.gridy=1;
		c.gridwidth=1;
		c.anchor=GridBagConstraints.PAGE_END;
		btnSave = new JButton("Guardar");
		btnSave.addActionListener(this);
		add(btnSave,c);

		pack();
	}

	public ConverterFrame(Converter converter){
		this();
		this.converter = converter;
	}

	@Override
	public void actionPerformed(ActionEvent e){
		if(e.getSource()==btnSave){

			JFileChooser fc = new JFileChooser();
			if(fc.showSaveDialog(this)==JFileChooser.APPROVE_OPTION){
				File file = fc.getSelectedFile();

				converter.convert(((WorkPanel)tabbedPane.getSelectedComponent()).getText());
				if(!converter.save(file)){
					JOptionPane.showMessageDialog(this,"Error","El archivo no ha sido almacenado.",JOptionPane.ERROR_MESSAGE);
				}else{
					JOptionPane.showMessageDialog(this,"Exito","El archivo ha sido almacenado adecuadamente.",JOptionPane.INFORMATION_MESSAGE);
				}
			}
		}
	}

	public static void main(String args[]){
		ConverterFrame mainFrame = new ConverterFrame();
		mainFrame.setVisible(true);
	}
}